samples/cellular/nrf_cloud_multi_service/Kconfig

```diff
@@ -285,6 +285,12 @@ config COAP_SHADOW_THREAD_STACK_SIZE
 
 endif
 
+config COAP_ALWAYS_CONFIRM
+	bool "Always use CON messaging when given a choice"
```

Comment:
i was confused by the qualifiers here... i assume they are used because logs and alerts are always sent as confirmable (which is hidden from the user).

something like 
`config COAP_SEND_CONFIRMABLE` 
`bool "Send device messages as confirmable"`

would be more simpler and less ambiguous.

Reply:
The AGNSS, PGPS, groundfix, FOTA, and shadow transfers are always CON. Previous to this PR, alerts and logs were NON; I made them CON now.

All other messages (sensor data, GNSS PVT, or custom messages sent with one of the `nrf_cloud_coap_*_send` functions) were NON; now they can be controlled.

Reply:
This Kconfig is used in the sample to set the default value for these `nrf_cloud_coap_*_send` functions' new confirmable parameters.

Reply:
In MSS context, the function impacted by this kconfig `send_device_message`.
So it would be helpful to add clarity to what "when given a choice" means.
The other stuff is done behind the scenes in nrf_cloud libraries (no choice).

Reply:
I took your suggested change, @jayteemo